vicare/climate.py

```diff
@@ -144,35 +135,25 @@ def target_temperature(self):
         return self._target_temperature
 
     @property
-    def current_operation(self):
-        """Return current operation ie. heat, cool, idle."""
-        if self._current_program == VICARE_PROGRAM_COMFORT:
-            return STATE_HEAT
-        elif self._current_program == VICARE_PROGRAM_ECO:
-            return STATE_ECO
-        elif self._current_program == VICARE_PROGRAM_NORMAL:
-            return STATE_AUTO
-        elif self._current_program == VICARE_PROGRAM_REDUCED:
-            return STATE_AUTO
-        elif self._current_program == VICARE_PROGRAM_STANDBY:
-            return STATE_OFF
-        else:
-            return VALUE_UNKNOWN
-
-    def set_operation_mode(self, operation_mode):
-        if operation_mode in self._operation_list:
-            """ 1st deactivate current mode """
-            if self._current_program != STATE_AUTO:
-                self._api.deactivateProgram(self._current_program)
+    def hvac_mode(self):
+        """Return current hvac mode"""
+        if self._current_program == VICARE_PROGRAM_STANDBY:
+            return HVAC_MODE_OFF
+        if self._current_mode == VICARE_MODE_DHWANDHEATING or self._current_mode == VICARE_MODE_DHW:
+            return HVAC_MODE_AUTO
+        return HVAC_MODE_OFF
+
+    def set_hvac_mode(self, hvac_mode):
+        if hvac_mode in self._hvac_modes:
+            """ 1st deactivate any existing program """
+            self._api.deactivateProgram(self._current_program)
```

Comment:
Well in case anyone sets an unsupported Hvac mode we should not handle this request and instead log an error:
            _LOGGER.error(
                "An error occurred while setting operation mode. "
                "Invalid operation mode: %s", operation_mode)

I guess this could happen when you set a mode e.g. via Home Assistant Service panel. The usual Lovelace UI should only show the supported hvac_modes